test/end-to-end/stock/stock.aggregate_consumption.page.js

```diff
@@ -0,0 +1,164 @@
+/* global element, by */
+const FU = require('../shared/FormUtils');
+const GU = require('../shared/GridUtils');
+const components = require('../shared/components');
+const SharedStockPage = require('./stock.shared.page');
+
+function StockAggregateConsumptionPage() {
+  const page = this;
+  const gridId = 'aggregated-consumption-grid';
+  const modalGridId = 'lots-grid';
+
+  // change current depot
+  page.changeDepot = SharedStockPage.setDepot;
+
+  // the grid id
+  page.gridId = gridId;
+  page.modalGridId = modalGridId;
+
+  // create modal
+  page.showCreateModal = async isNewRequisition => {
+    const selector = isNewRequisition ? '[data-method="create-record"]' : '[data-method="create-other-record"]';
+    await $(selector).click();
+  };
+
+  page.showSearchModal = () => {
+    return $('[data-method="search"]').click();
+  };
+
+  page.setFiscalPeriod = async (fiscalYear, period) => {
+    await components.fiscalYearPeriodSelect.set(fiscalYear, period);
+  };
+
+  // depot select
+  page.setDepot = depot => {
+    return components.depotSelect.set(depot, 'depot-supplier');
+  };
+
+  page.setRows = async number => {
+    await components.addItem.set(number);
+  };
+
+  // set description
+  page.setDescription = description => {
+    return FU.input('StockCtrl.movement.description', description);
+  };
+
+  /**
+   * @method setHeaderValue
+   */
+  page.setHeaderValue = async function setHeaderValue(rowIndex, columnIndex, value) {
+    const cell = await GU.getCell(gridId, rowIndex, columnIndex);
+    const inputOutStock = cell.element(by.id('row.treeNode.children[0].row.entity.inventory_uuid'));
```

Comment:
This: `by.id('row.treeNode.children[0].row.entity.inventory_uuid')`  selector makes me think something is broken.  Shouldn't it be replaced by the inventory uuid?

I also suggest that you use the inventory text instead of the `id=inventory_uuid`.  Set something like `data-inventory-label="{{row.treeNode.children[0].row.entity.inventory_text}}" which will let you change this selector to:

```js
const inputOutStock = cell.$('[data-inventory-label="Amoxy ..."]')); 
```
which is much easier to read.

Reply:
If I made this modification, I will also be forced to modify the server part to do the reverse operation to find the Uuid from the name of the inventory